pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -243,5 +249,9 @@ func (p *Provider) getVnetInfoLabels(subnetID string) (map[string]string, error)
 		azureCNIOverlayLabel: strconv.FormatBool(true),
 		podNetworkTypeLabel:  consts.NetworkPluginModeOverlay,
 	}
+
+	parsedVersion, _ := semver.Parse(kubernetesVersion)
```

Comment:
I think you need light modification/safety here.

See `UseAzureLinux3`
https://github.com/Azure/karpenter-provider-azure/blob/38368736f1304b6156443367bcc9ced16cdcd006/pkg/providers/imagefamily/utils.go#L25
```
	// Parse version, stripping any 'v' prefix if present
	version, err := semver.ParseTolerant(strings.TrimPrefix(kubernetesVersion, "v"))
	if err != nil {
		// If we can't parse the version, default to AzureLinux (false)
		return false
	}
```

At least stripping the `v` I think is required. I think we prefix with `v`, and semver treats that as an error.

Alex recently added the `ParseTolerant`, instead of `Parse` as well.

Reply:
Looked into it further, and the prefix should be trimmed by the k8s provider, and set as such without it by controller. However, I still think its at least work checking the potential error, rather than discarding it `_`, and as such, might as well be defensive with the same parsing as `UseAzureLinux3`

Reply:
yeah i originally had the error handling that if k8s version parsing failed we would return an error, i can add this back and use ParseTolerant